portal/src/main/scala/cromwell/pipeline/controller/ProjectController.scala

```diff
@@ -0,0 +1,53 @@
+package cromwell.pipeline.controller
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.{ as, complete, delete, entity, get, onComplete, path, pathPrefix, _ }
+import akka.http.scaladsl.server.PathMatchers.JavaUUID
+import akka.http.scaladsl.server.Route
+import cromwell.pipeline.datastorage.dto.{ ProjectAdditionRequest, ProjectId }
+import cromwell.pipeline.service.ProjectService
+import cromwell.pipeline.utils.auth.AccessTokenContent
+import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport._
+
+import scala.concurrent.ExecutionContext
+import scala.language.{ implicitConversions, postfixOps }
+import scala.util.Success
+
+class ProjectController(projectService: ProjectService)(implicit executionContext: ExecutionContext) {
+
+  val route: AccessTokenContent => Route = accessToken =>
+    pathPrefix("projects") {
+      post {
+        entity(as[ProjectAdditionRequest]) { request =>
+          onComplete(projectService.addProject(request)) {
+            case Success(projectId) => complete(projectId.value)
+            case _                  => complete(StatusCodes.InternalServerError)
+          }
+        }
+      } ~
+        path(JavaUUID) { uuid =>
+          get {
+            onComplete(projectService.getProjectById(ProjectId(uuid.toString))) {
+              case Success(Some(project)) => complete(project)
+              case Success(None)          => complete(StatusCodes.NotFound, "Project not found")
+              case _                      => complete(StatusCodes.BadRequest)
+            }
+          } ~
+            delete {
+              onComplete(projectService.getProjectById(ProjectId(uuid.toString))) {
```

Comment:
Working with `User` entity we decided that `DELETE` endpoint is not related to the uuid. 
You need to sent full Project entity. And project id should be taken from there.

Reply:
we discussed it, delete/{UUID} is fine